runtime/parser.go

```diff
@@ -157,6 +164,54 @@ func (r *CmdConfig) selectConfigSetByName(name string) (Runtime, error) {
 	return r.ConfigSets[name], nil
 }
 
+func (c *AllCmdConfigs) Add(pkgName string, cmdConfig *CmdConfig) {
+	if c.cmdConfigs == nil {
+		c.cmdConfigs = make(map[string]*CmdConfig, 15)
+	}
+	c.cmdConfigs[pkgName] = cmdConfig
+	c.order = append(c.order, pkgName)
+}
+
+func (c *AllCmdConfigs) Persist(mpmDir string) error {
+	// Prepare directory to store bootcmd files in.
+	targetDir := filepath.Join(mpmDir, "run")
+	if _, err := os.Stat(targetDir); err != nil {
+		if err = os.MkdirAll(targetDir, 0775); err != nil {
+			return err
+		}
+	}
+
+	// Persist runscript scripts for all config_sets of all packages.
+	for _, pkgName := range c.order {
+		cmdConf := c.cmdConfigs[pkgName]
+		if cmdConf == nil {
+			continue
+		}
+
+		for confName := range cmdConf.ConfigSets {
+			currConf := cmdConf.ConfigSets[confName]
+			// Validate.
+			if err := currConf.Validate(); err != nil {
+				return fmt.Errorf("Validation failed for configuration set '%s': %s", confName, err)
+			}
+
+			// Calculate boot command.
+			bootCmd, err := currConf.GetBootCmd(c.cmdConfigs)
+			if err != nil {
+				return err
+			}
+
+			// Persist to file.
+			cmdFile := filepath.Join(targetDir, confName)
+			if err := ioutil.WriteFile(cmdFile, []byte(bootCmd), 0775); err != nil {
```

Comment:
Do you think we should make this `500`? In particular, I wonder whether we should remove the `w` flag in order to prevent file overrides?